File: solutions/de.vimotest.sandbox/source_gen/de/vimotest/sandbox/widgetassertions/CheckBoxes_ViewModel.cs
/// <filename>
///     CheckBoxes_ViewModel.cs
/// </filename>


namespace widgetassertions
{
    public class CheckBoxes_ViewModel
  {
        private bool isMyFlagWithLabelCheckBoxChecked;
     
        public     virtual bool getIsMyFlagWithLabelCheckBoxChecked()
    {
      return this.isMyFlagWithLabelCheckBoxChecked;
    }
     
        private bool isMyFlagWithLabelCheckBoxVisible;
     
        public     virtual bool getIsMyFlagWithLabelCheckBoxVisible()
    {
      return this.isMyFlagWithLabelCheckBoxVisible;
    }
     
        private bool isMyFlagWithLabelCheckBoxEnabled;
     
        public     virtual bool getIsMyFlagWithLabelCheckBoxEnabled()
    {
      return this.isMyFlagWithLabelCheckBoxEnabled;
    }
     
        private string MyFlagWithLabelCheckBoxText;
     
        public     virtual string getMyFlagWithLabelCheckBoxText()
    {
      return this.MyFlagWithLabelCheckBoxText;
    }
     
        private bool isMyFlagNoLabelCheckBoxChecked;
     
        public     virtual bool getIsMyFlagNoLabelCheckBoxChecked()
    {
      return this.isMyFlagNoLabelCheckBoxChecked;
    }
     
        private bool? isMyFlagTriStateCheckBoxChecked;
     
        public     virtual bool? getIsMyFlagTriStateCheckBoxChecked()
    {
      return this.isMyFlagTriStateCheckBoxChecked == true;
    }
  }
}

File: solutions/de.vimotest.sandbox/source_gen/de/vimotest/sandbox/widgetassertions/CheckBoxes_ViewModel.cpp
#include "CheckBoxes_ViewModel.h"
#include <optional>
#include <string>

namespace widgetassertions
{
  bool CheckBoxes_ViewModel::getIsMyFlagWithLabelCheckBoxChecked()
  {
    return this->isMyFlagWithLabelCheckBoxChecked;
  }
  bool CheckBoxes_ViewModel::getIsMyFlagWithLabelCheckBoxVisible()
  {
    return this->isMyFlagWithLabelCheckBoxVisible;
  }
  bool CheckBoxes_ViewModel::getIsMyFlagWithLabelCheckBoxEnabled()
  {
    return this->isMyFlagWithLabelCheckBoxEnabled;
  }
  std::string CheckBoxes_ViewModel::getMyFlagWithLabelCheckBoxText()
  {
    return this->MyFlagWithLabelCheckBoxText;
  }
  bool CheckBoxes_ViewModel::getIsMyFlagNoLabelCheckBoxChecked()
  {
    return this->isMyFlagNoLabelCheckBoxChecked;
  }
  std::optional<bool> CheckBoxes_ViewModel::getIsMyFlagTriStateCheckBoxChecked()
  {
    return this->isMyFlagTriStateCheckBoxChecked.value() == true;
  }
}
